dora/shell/src/main/java/alluxio/cli/fsadmin/report/CapacityCommand.java

```diff
@@ -296,96 +309,58 @@ private void printWorkerAllMasterConnectionInfo(
    *
    * @param workerInfoList the worker info list to get info from
    */
-  private void printWorkerInfo(List<WorkerInfo> workerInfoList) {
-    mIndentationLevel = 0;
+  private void generateWorkerInfo(List<WorkerInfo> workerInfoList) {
+    boolean isShort = false;
     if (mCapacityTierInfoMap.size() == 0) {
       return;
     } else if (mCapacityTierInfoMap.size() == 1) {
-      // TODO(jiacheng): test BOTH long and short output
-      // Do not print Total value when only one tier exists
-      printShortWorkerInfo(workerInfoList);
-      return;
+      isShort = true;
     }
+
+    ArrayNode workerInfo = mMapper.createArrayNode();
     Set<String> tiers = mCapacityTierInfoMap.keySet();
-    String tiersInfo = String.format(Strings.repeat("%-14s", tiers.size()), tiers.toArray());
-    String longInfoFormat = getInfoFormat(workerInfoList, false);
-    print(String.format("%n" + longInfoFormat,
-        "Worker Name", "State", "Last Heartbeat", "Storage", "Total", tiersInfo,
-        "Version", "Revision"));
 
     for (WorkerInfo info : workerInfoList) {
-      String workerName = info.getAddress().getHost();
-
-      long usedBytes = info.getUsedBytes();
-      long capacityBytes = info.getCapacityBytes();
-
-      String usedPercentageInfo = "";
-      if (capacityBytes != 0) {
-        int usedPercentage = (int) (100L * usedBytes / capacityBytes);
-        usedPercentageInfo = String.format(" (%s%%)", usedPercentage);
+      ObjectNode infoPerWorker = mMapper.createObjectNode();
+
+      ArrayNode tiersInfo = mMapper.createArrayNode();
+
+      if (isShort) {
+        // TODO(jiacheng): test BOTH long and short output
+        // Do not print Total value when only one tier exists
+        ObjectNode tierInfo = mMapper.createObjectNode();
+        tierInfo.put("Tier", mCapacityTierInfoMap.firstKey());
+        tierInfo.put("Capacity", FormatUtils.getSizeFromBytes(info.getCapacityBytes()));
+        tierInfo.put("Used", FormatUtils.getSizeFromBytes(info.getUsedBytes()));
+        if (info.getCapacityBytes() != 0) {
+          tierInfo.put("Used percentage", String.format(" %s%%", (int) (100L * info.getUsedBytes() / info.getCapacityBytes())));
```

Comment:
do you still need this prefixing space character?
```suggestion
          tierInfo.put("Used percentage", String.format("%s%%", (int) (100L * info.getUsedBytes() / info.getCapacityBytes())));
```